protocol/testing/e2e/gov/vest_test.go

```diff
@@ -0,0 +1,254 @@
+		updateExistingVestEntry  bool // whether above msg should update an existing vest entry.
+		expectCheckTxFails       bool
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success: create a new vest entry": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   "random_vester",
+					TreasuryAccount: "random_treasury",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 2, 0, 0, 0, 0, time.UTC),
+					EndTime:         time.Date(2024, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Success: update an existing vest entry": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   "random_vester",
+					TreasuryAccount: "random_treasury",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 2, 0, 0, 0, 0, time.UTC),
+					EndTime:         time.Date(2024, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			updateExistingVestEntry: true,
+			expectedProposalStatus:  govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Failure: vester account is empty": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   "",
+					TreasuryAccount: "random_treasury",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 2, 0, 0, 0, 0, time.UTC),
+					EndTime:         time.Date(2024, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectCheckTxFails: true,
+		},
+		"Failure: treasury account is empty": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   "random_vester",
+					TreasuryAccount: "",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 2, 0, 0, 0, 0, time.UTC),
+					EndTime:         time.Date(2024, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectCheckTxFails: true,
+		},
+		"Failure: start time after end time": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   "random_vester",
+					TreasuryAccount: "",
+					Denom:           "dv4tnt",
+					StartTime:       time.Date(2023, 10, 1, 0, 0, 0, 1, time.UTC),
+					EndTime:         time.Date(2023, 10, 1, 0, 0, 0, 0, time.UTC),
+				},
+			},
+			expectCheckTxFails: true,
+		},
+		"Failure: invalid authority": {
+			msg: &vesttypes.MsgSetVestEntry{
+				Authority: constants.BobAccAddress.String(),
+				Entry: vesttypes.VestEntry{
+					VesterAccount:   vesttypes.ModuleName,
```

Comment:
Nit: `vesttypes.ModuleName` is never used as a vester account so it's bit misleading how it's used here (See [rewartypes.VesterAccountName](https://github.com/dydxprotocol/v4-chain/blob/604ba626f00be9b7a145b702fee3e2dea42727ce/protocol/x/rewards/types/keys.go#L33C15-L33C15) or [CommunityVesterAccountName](https://github.com/dydxprotocol/v4-chain/blob/604ba626f00be9b7a145b702fee3e2dea42727ce/protocol/x/vest/types/keys.go#L30))